### PR TITLE
Update nfr and glossary terms

### PR DESCRIPTION
**Pull request by derekjxtan**
Added more non functional requirements.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -485,16 +485,21 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+6. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
 *{More to be added}*
 
 ### Glossary
 
 * **Mainstream OS**: Windows, Linux, Unix, OS-X
+* **Patient**: A person receiving medical services at UHC
+* **Doctor**: A person providing medical services at UHC
+* **Appointment**: A arranged meeting between a patient and a doctor at a particular time
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:502]: Change "time" to "date and time"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -485,16 +485,21 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1. Should work on any _mainstream OS_ as long as it has Java `11` or above installed.
+2. Should be able to handle at least 60,000 patients at any point of time.
+3. Should be able to add, view, edit and delete patients and doctors without a noticeable lag.
+4. Should be able to retrieve all appointments for a particular doctor or patient within 1 second.
+5. Should lose no more than 1-minute worth of data in case of system error. 
+6. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
 *{More to be added}*
 
 ### Glossary
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:497]: Terms that could be added "Receptionist" and "NRIC"

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (derekjxtan) [reply docs/DeveloperGuide.md]: changed

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (derekjxtan) [reply docs/DeveloperGuide.md:497]: added

**Pull request merged**